File: 2022_KAKAO_BLIND_RECRUITMENT/secret_map/solution.java
import java.util.*;

class Solution {
    public String leftPad(String str, int padNum, String pattern) {
        int strLength = str.length();
        
        if (strLength < padNum) {
            for (int i=0; i<padNum - strLength; i++) {
                str = pattern + str;
            }
        }
        
        return str;     
    }
    
    public String[] solution(int n, int[] arr1, int[] arr2) {
        String[] answer = new String[n];
        
        for (int i=0; i<n; i++) {
            String map = "";
            char[] ch1 = leftPad(Integer.toBinaryString((int)arr1[i]), n, "0").toCharArray();
            char[] ch2 = leftPad(Integer.toBinaryString((int)arr2[i]), n, "0").toCharArray();
            
            for (int j=0;j<ch1.length;j++) {
                if ('1' == ch1[j] || '1' == ch2[j]) {
                    map = map + "#";
                } else {
                    map = map + " ";
                }
            }
            
            answer[i] = map;
        }
        
        return answer;
    }
}

File: 2022_KAKAO_BLIND_RECRUITMENT/secret_map/solution.js
function solution(n, arr1, arr2) {
    const answer = [];

    for (let i=0; i<n; i++) {
        const final_map = []
        const map1 = Array.from(arr1[i].toString(2).padStart(n,0))
        const map2 = Array.from(arr2[i].toString(2).padStart(n,0))

        map1.map((val, idx) => {
            if (val === '1' || map2[idx] === '1') {
                final_map.push('#')
            } else {
                final_map.push(' ')
            }
        })

        answer.push(final_map.join(''))
    }

    return answer;
}